src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1738,6 +1796,61 @@ skip:
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  void t_cryptonote_protocol_handler<t_core>::select_peer_for_testing(uint64_t block_height, const cryptonote::blobdata &block, const boost::uuids::uuid &except, std::vector<boost::uuids::uuid> &selected)
+  {
+    std::vector<std::pair<boost::uuids::uuid, int64_t>> peers;
+    m_p2p->for_each_connection([&](cryptonote_connection_context& ctx, nodetool::peerid_type peer_id, uint32_t support_flags)->bool{
+      if (ctx.m_connection_id == except)
+        return true;
+      if (ctx.m_remote_blockchain_height < block_height)
+        return true;
+      if (ctx.m_state < cryptonote_connection_context::state_normal)
+        return true;
+      if (ctx.m_waiting_for_block != "")
+        return true;
+      if (ctx.m_remote_address.get_zone() != epee::net_utils::zone::public_)
+        return true;
+      peers.push_back(std::make_pair(ctx.m_connection_id, ctx.m_score));
+      return true;
+    });
+    if (peers.size() <= 2)
+      return;
+    const size_t n_tests = 1 + peers.size() / 9;
```

Comment:
We changed default peers to 12. Would you prefer `1 + peers.size() / 9` or `1 + peers.size() / 13` ?

Reply:
1 + peers.size() / 9